### PR TITLE
Add Konflux client, event-sender plugin and must-gather images to CSV

### PR DESCRIPTION
**Pull request by creydr**
Adds the Konfluxs builds of :
* kn-client
* event-sender plugin and
* must-gather 

to the CSV

### CODE REVIEW

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -935,7 +935,7 @@ spec:
                       - name: "IMAGE_KN_PLUGIN_EVENT_SENDER"
                         value: "registry.ci.openshift.org/knative/release-1.15:client-plugin-event-sender"
                       - name: "IMAGE_KN_CLIENT"
-                        value: "registry.ci.openshift.org/openshift/knative-v1.15:knative-client-kn"
+                        value: "registry.redhat.io/openshift-serverless-1/kn-client-kn-rhel8@"
```
Comment (creydr) [inline olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:938]: the reason, why it is on-hold/WIP

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -933,9 +933,9 @@ spec:
                       - name: "IMAGE_KUBE_RBAC_PROXY"
                         value: "registry.redhat.io/openshift4/ose-kube-rbac-proxy-rhel9@sha256:6fbeaf03058bb5d6e9e3311a7afe67666e0770532c978b6b57e77ea2851cb085"
                       - name: "IMAGE_KN_PLUGIN_EVENT_SENDER"
-                        value: "registry.ci.openshift.org/knative/release-1.15:client-plugin-event-sender"
+                        value: "registry.redhat.io/openshift-serverless-1/kn-plugin-event-sender-rhel8@"
```
Comment (creydr) [inline olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:936]: same as https://github.com/openshift-knative/serverless-operator/pull/2983#discussion_r1824470825

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (creydr) [reply olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml]: done

File: hack/lib/images.bash [hunk 3]
```diff
@@ -211,6 +212,43 @@ function knative_kn_plugin_func_images() {
...
+
+function knative_kn_plugin_event_images() {
+  local knative_kn_plugin_event tag app_version
+  tag=${1:?"Provide tag for kn-plugin-event images"}
+
+  app_version=$(get_app_version_from_tag "${tag}")
+  knative_kn_plugin_event="${registry_prefix_quay}${app_version}/kn-plugin-event"
+
+  export KNATIVE_KN_PLUGIN_EVENT_SENDER=${KNATIVE_KN_PLUGIN_EVENT_SENDER:-$(latest_registry_redhat_io_image_sha "${knative_kn_plugin_event}-sender:${tag}")}
+}
```
Comment (creydr) [inline hack/lib/images.bash:231]: maybe we can merge this with `knative_kn_plugin_func` :shrug:

Comment (mgencur) [reply hack/lib/images.bash:231]: I would keep it separate.

**Comment by cardil** [comment]
+1 lgtm

**Comment by creydr** [comment]
```
E1107 14:18:57.459171      23 memcache.go:265] couldn't get current server API group list: Get "https://api.serverless-ocp-4-17-amd64-aws-us-east-1-hmjvv.serverless.devcluster.openshift.com:6443/api?timeout=32s": dial tcp: lookup api.serverless-ocp-4-17-amd64-aws-us-east-1-hmjvv.serverless.devcluster.openshift.com on 172.30.0.10:53: no such host
Unable to connect to the server: dial tcp: lookup api.serverless-ocp-4-17-amd64-aws-us-east-1-hmjvv.serverless.devcluster.openshift.com on 172.30.0.10:53: no such host
```
/test 417-upstream-e2e-aws-417

**Comment by creydr** [comment]
```
ERRO[2024-11-08T09:54:56Z] 
  * could not run steps: step serverless-source-image failed: error occurred handling build serverless-source-image-amd64: could not get build serverless-source-image-amd64: builds.build.openshift.io "serverless-source-image-amd64" not found 
```
/retest

**Comment by creydr** [comment]
```
sh-4.4$ /cli/oc -n openshift-serverless get po
NAME                                         READY   STATUS                  RESTARTS   AGE
knative-openshift-5f9657c985-c844w           0/1     Init:ImagePullBackOff   0          4m9s
...
```

```
  initContainerStatuses:
  - image: registry.redhat.io/openshift-serverless-1/kn-client-cli-artifacts-rhel8@sha256:82a41eaae2cd8b6004dcb663a2cd359c09fecf646bf5beaec330463abea0ea90
    imageID: ""
    lastState: {}
    name: cli-artifacts
    ready: false
    restartCount: 0
    started: false
    state:
      waiting:
        message: Back-off pulling image "registry.redhat.io/openshift-serverless-1/kn-client-cli-artifacts-rhel8@sha256:82a41eaae2cd8b6004dcb663a2cd359c09fecf646bf5beaec330463abea0ea90"
        reason: ImagePullBackOff
```

We don't have it in the ICSP, because it's not part of the csv and thus not in the bundle yet:
```
$ /cli/oc get imagecontentsourcepolicies.operator.openshift.io serverless-image-content-source-policy | grep -i "artifacts"
$  
```

**Pull request merged**